src/network/networkd-ndisc.c

```diff
@@ -193,6 +193,11 @@ static int ndisc_request_route(Route *in, Link *link, sd_ndisc_router *rt) {
                 r = sd_ndisc_router_get_mtu(rt, &mtu);
                 if (r < 0 && r != -ENODATA)
                         return log_link_warning_errno(link, r, "Failed to get default router MTU from RA: %m");
+
+                link->network->ipv6_hop_limit = hop_limit;
```

Comment:
hop_limit is always zero here?

also, the network objects carry just the data from the .network file, and shouldn't patched around in, so this doesn't look right.

Reply:
Ohh My bad fixed. Thanks !